solutions/joi-19-cake-circle.mdx

```diff
@@ -0,0 +1,180 @@
+---
+id: joi-19-cake-circle
+source: JOI 2019
+title: Cake Circle
+author: Dustin Miao
+---
+
+# Explanation
+
+First, given some set of cakes $S$, it is always optimal to arrange them in sorted order. The total color is $2(\max(S) - \min(S))$. 
+
+Sort the cakes by color. The task now becomes choosing some range $[l, r]$ that maximizes the sum of the $m$ largest beauty values, minus $2 \cdot (C_r - C_l)$.
```

Comment:
> The task now becomes choosing some range $[l, r]$ that maximizes the sum of the $m$ largest beauty values, minus $2 \cdot (C_r - C_l)$.

how??

Reply:
It's in sorted order now, so we just greedily take the largest values and subtract two times the range (from above)

Reply:
so tell them that